doc/mgmt/Docker to Host communication.md

```diff
@@ -0,0 +1,204 @@
+# Feature Name
+Docker to Host communication
+
+# High Level Design Document
+#### Rev 0.1
+
+# Table of Contents
+  * [List of Tables](#list-of-tables)
+  * [Revision](#revision)
+  * [About This Manual](#about-this-manual)
+  * [Scope](#scope)
+  * [Definition/Abbreviation](#definitionabbreviation)
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 10/28/2019  | Nirenjan Krishnan  | Initial version                   |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.2 | 12/08/2019  | Mike Lazar         | Add details about architecture    |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.3 | 12/16/2019  | Mike Lazar         | Add security and logging info     |
+
+
+# About this Manual
+This document provides general information about the Docker to Host
+communication feature in SONiC.
+
+# Scope
+This document describes the high level design of Docker to Host communication.
+This describes the infrastructure provided by the feature, and example usage,
+however, it does not describe the individual host-specific features.
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+| **Term**                 | **Meaning**                                       |
+|--------------------------|---------------------------------------------------|
+| D-Bus                    | Desktop Bus: https://en.wikipedia.org/wiki/D-Bus  |
+
+# 1 Feature Overview
+
+The management framework runs within a Docker container, and performs actions
+translating the user CLIs or REST requests to actions. Most of these actions
+perform some operation on the Redis database, but some of them require
+operations to be done on the host, i.e., outside the container. This document
+describes the host server, and translib API that are used to communicate between
+the Docker container and the host.
```

Comment:
It's greatly useful. However, are we restricting the sonic users to use MGMT container to enjoy the DBUS framework? Let's say from xyz container user wanted to communicate to Host, user should be able to use it as a plug a play?

Reply:
As mentioned in the document, any container that mounts /var/run/dbus (-v run/create docker option) can use Linux host D-Bus services in general - any pre-existing D-Bus services,  and not just "Host" services developed for the management container. Using /var/run/dbus is a choice of the container developer, and it is completely outside the scope of this feature. D-Bus is an existing Linux feature, and bind mount (-v option) a docker feature.

Reply:
Thanks Mike. Today we should be able to program most of the switch configuration through redis right? Can you shed some light on what are the use cases/configurations are trying to solve using   D-Bus?

Reply:
Given the talk about security and DBus and Containers - it's worth having a conversation with the flatpak community who have already implemented this called 'portals' of which we do consider security implementations. Happy to connect.